src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/check_health.py

```diff
@@ -0,0 +1,428 @@
+
+
+class Errors(enum.Enum):
+    DOCKER_DAEMON_ERROR = 1
+    DOCKER_VERSION_ERROR = 2
+    ACR_CLI_VERSION_ERROR = 4
+    HELM_CLIENT_VERSION_ERROR = 8
+    HELM_SERVER_VERSION_ERROR = 16
+    LOGIN_SERVER_ERROR = 32
+    CONNECTIVITY_REGISTRY_STATUS_ERROR = 64
+    CONNECTIVITY_ENDPOINT_FORBIDDEN_ERROR = 128
+    CONNECTIVITY_ENDPOINT_CHALLENGE_ERROR = 256
+    CONNECTIVITY_AAD_LOGIN_ERROR = 512
+    CONNECTIVITY_REFRESH_TOKEN_ERROR = 1024
+    CONNECTIVITY_ACCESS_TOKEN_ERROR = 2048
+    DOCKER_PULL_ERROR = 4096
+    DOCKER_COMMAND_ERROR = 8192
+    HELM_COMMAND_ERROR = 16384
+
+
+recommendations_map = {
+    "DOCKER_DAEMON_ERROR": "restart Docker daemon",
+    "DOCKER_VERSION_ERROR": "update Docker version",
+    "ACR_CLI_VERSION_ERROR": "update Azure CLI version",
+    "HELM_CLIENT_VERSION_ERROR": "update helm version",
+    "HELM_SERVER_VERSION_ERROR": "update helm version; restart kubernetes manager",
+    "LOGIN_SERVER_ERROR": "check spelling and permissions on the registry; run az login to refresh token",
+    "CONNECTIVITY_REGISTRY_STATUS_ERROR": "check if registry exists, if name is correct and check permissions on it",
+    "CONNECTIVITY_ENDPOINT_FORBIDDEN_ERROR": "add current IP to network rules",
+    "CONNECTIVITY_ENDPOINT_CHALLENGE_ERROR": "contact ACR team",
+    "CONNECTIVITY_AAD_LOGIN_ERROR": "enable AAD login (contact ACR team)",
+    "CONNECTIVITY_REFRESH_TOKEN_ERROR": "run az login to refresh token",
+    "CONNECTIVITY_ACCESS_TOKEN_ERROR": "check permissions to the registry; run az login to refresh token",
+    "DOCKER_PULL_ERROR": "restart docker services",
+    "DOCKER_COMMAND_ERROR": "install, start or restart docker client and services",
+    "HELM_COMMAND_ERROR": "install or start helm client; restart kubernetes manager"
+}
+
+
+## Utilities functions ##
+
+
+def print_pass(message):
+    from colorama import Fore, Style, init
+    init()
+    print(str(message) + " : " + Fore.GREEN + "OK" + Style.RESET_ALL)
+
+
+def print_warning(message):
+    from colorama import Fore, Style, init
+    init()
+    print(Fore.YELLOW + str(message) + Style.RESET_ALL)
+
+
+def print_error(message):
+    from colorama import Fore, Style, init
+    init()
+    print(Fore.RED + str(message) + Style.RESET_ALL)
+
+
+def _error_handler(error, operation, ignore_errors, additional_message=None):
+    error_message = \
+        "{} : FAILED\n" \
+        "Error : {}\n" \
+        "Recommendations : {}" \
+        .format(operation, error.name, recommendations_map[error.name])
+
+    if additional_message:
+        error_message += "\n" + additional_message
+
+    if ignore_errors:
+        print_error(error_message)
+        return error.value
+    else:
+        _raise_exception(error.value, error_message)
+
+
+def _raise_exception(accumulated_errors, error_message=None):
+    if accumulated_errors == 0:
+        print("All checks passed. Please refer to https://aka.ms/acr/faq for more informations.")
+        return None
```

Comment:
Just return?

Reply:
Every command must return an object or None. Since this is the last method called, we should return None

Reply:
I will change the name of this method to "_handle_exception" as we don't always raise the exception.